frontend/src/app/modules/job-result/job-result.component.ts

```diff
@@ -0,0 +1,322 @@
+        (event.target as HTMLSpanElement).matches(dayElementsInMonthView)
+      ) {
+        if (dateTimeComponent === DateTimeRange.FROM) {
+          this.dateTimeRangeFrom.close();
+          this.dateTimeRangeTo.open();
+        } else if (dateTimeComponent === DateTimeRange.TO) {
+          this.dateTimeRangeTo.close();
+        }
+      }
+    });
+  }
+
+  private setCalendarLanguage(): void {
+    if (this.osmLangService.getOsmLang() === 'en') {
+      this.dateTimeAdapter.setLocale('en-GB');
+    } else {
+      this.dateTimeAdapter.setLocale(this.osmLangService.getOsmLang());
+    }
+  }
+
+  private getAllJobs(): void {
+    this.dataService.getAllJobs()
+      .subscribe((jobs: Job[]) => {
+        this.jobs = jobs;
+        this.readQueryParams();
+      });
+  }
+
+  private readQueryParams(): void {
+    this.route.queryParams.subscribe((params: Params) => {
+      if (this.checkQuery(params)) {
+        const jobId = parseInt(decodeURIComponent(params.job), 10);
+        if (jobId) {
+          this.selectedJob = this.jobs.find((job: Job) => job.id === jobId);
+          this.filter.dateTimeRange = [new Date(decodeURIComponent(params.from)), new Date (decodeURIComponent(params.to))];
+          this.filter.testAgent = params.testAgent ? decodeURIComponent(params.testAgent) : '';
+          this.filter.jobResultStatus = this.statusService.readStatusByQueryParam(
+            params.status, JobResultStatus, this.statusService.JOB_RESULT_STATUS_GROUPS
+          );
+          this.filter.wptStatus = this.statusService.readStatusByQueryParam(
+            params.wptStatus, WptStatus, this.statusService.WPT_STATUS_GROUPS
+          );
+          this.filter.description = params.description ? decodeURIComponent(params.description) : '';
+          this.getJobResults(jobId);
+        }
+      }
+    }).unsubscribe();
+  }
+
+  private writeQueryParams(): void {
+    const params: Params = {
+      job: this.selectedJob ? encodeURIComponent(this.selectedJob.id.toString(10)) : null,
+      from: this.selectedJob && this.filter.dateTimeRange[0] && !isNaN(this.filter.dateTimeRange[0].valueOf()) ?
+        encodeURIComponent(this.filter.dateTimeRange[0].toISOString()) :
+        null,
+      to: this.selectedJob && this.filter.dateTimeRange[1] && !isNaN(this.filter.dateTimeRange[1].valueOf()) ?
+        encodeURIComponent(this.filter.dateTimeRange[1].toISOString()) :
+        null,
+      testAgent: this.selectedJob && this.filter.testAgent !== '' ? encodeURIComponent(this.filter.testAgent) : null,
+      status: this.statusService.writeStatusAsQueryParam(this.filter.jobResultStatus, JobResultStatus, !!this.selectedJob),
+      wptStatus: this.statusService.writeStatusAsQueryParam(this.filter.wptStatus, WptStatus, !!this.selectedJob),
+      description: this.selectedJob && this.filter.description !== '' ? encodeURIComponent(this.filter.description) : null
```

Comment:
Could you refactor this? Some functions for the fields would be nice to understand was is happening with the blink of an eye.